src/core/js/models/adaptModel.js

```diff
@@ -1,859 +1,853 @@
-      ].filter(Boolean);
-      const typeGroups = this.getTypeGroups();
-      if (_.intersection(pluralizedLowerCaseTypes, typeGroups).length) {
-        return true;
-      }
-      return false;
-    }
+    this.set({
+      _isComplete: true,
+      _isInteractionComplete: true
+    });
+  }
 
-    /**
-     * Returns an array of strings describing the model type groups.
-     * All strings are lowercase and should not be pluralized.
-     * i.e. ['course', 'menu', 'contentobject'], ['page', 'contentobject'], ['component']
-     * @returns {[string]}
-     */
-    getTypeGroups() {
-      if (this._typeGroups) return this._typeGroups;
-      const typeGroups = [ this.get('_type') ];
-      let parentClass = this;
-      while ((parentClass = Object.getPrototypeOf(parentClass))) {
-        if (!parentClass.hasOwnProperty('getTypeGroup')) continue;
-        typeGroups.push(parentClass.getTypeGroup.call(this));
-      }
-      return (this._typeGroups = _.uniq(typeGroups.filter(Boolean).map(s => s.toLowerCase())));
-    }
-
-    /**
-     * Searches the model's ancestors to find the first instance of the specified ancestor type
-     * @param {string} [ancestorType] Valid values are 'course', 'pages', 'contentObjects', 'articles' or 'blocks'.
-     * If left blank, the immediate ancestor (if there is one) is returned
-     * @return {object} Reference to the model of the first ancestor of the specified type that's found - or `undefined` if none found
-     */
-    findAncestor(ancestorType) {
-      const parent = this.getParent();
-      if (!parent) return;
-      if (!ancestorType || parent.isTypeGroup(ancestorType)) {
-        return parent;
-      }
-      return parent.findAncestor(ancestorType);
-    }
-
-    /**
-     * Returns all the descendant models of a specific type
-     * @param {string} descendants Valid values are 'contentobject', 'page', 'menu', 'article', 'block', 'component', 'question'
-     * @param {object} options an object that defines the search type and the properties/values to search on. Currently only the `where` search type (equivalent to `Backbone.Collection.where()`) is supported.
-     * @param {object} options.where
-     * @return {array}
-     * @example
-     * //find all available, non-optional components
-     * this.findDescendantModels('component', { where: { _isAvailable: true, _isOptional: false }});
-     */
-    findDescendantModels(descendants, options) {
-      const allDescendantsModels = this.getAllDescendantModels();
-      const returnedDescendants = allDescendantsModels.filter(model => {
-        return model.isTypeGroup(descendants);
-      });
+  checkCompletionStatus() {
+    // defer to allow other change:_isComplete handlers to fire before cascading to parent
+    Adapt.checkingCompletion();
+    _.defer(this.checkCompletionStatusFor.bind(this), '_isComplete');
+  }
 
-      if (!options) {
-        return returnedDescendants;
-      }
+  checkInteractionCompletionStatus() {
```

Comment:
This is not required but you could make a single function that carries out the tasks of `checkCompletionStatus` and `checkInteractionCompletionStatus` and pass through `_isComplete` and `_isInteractionComplete` as a parameter, e.g.:

```
checkCompletionStatus(competionType) {
    // defer to allow other change:_isComplete handlers to fire before cascading to parent
    Adapt.checkingCompletion();
    _.defer(this.checkCompletionStatusFor.bind(this), competionType);
  }
  ```

Reply:
I'm not going to change the API here, but you do have a good point. I'd have to put deprecated warnings in etc for the missing function.